core/src/main/codegen/templates/Parser.jj

```diff
@@ -1894,7 +1894,7 @@ SqlNode TableRef2(boolean lateral) :
             tableRef = unnestOp.createCall(s.end(this), args.toArray());
         }
     |
-        [ LOOKAHEAD(1) <LATERAL> { lateral = true; } ]
```

Comment:
This one is removed cause it is redundant:
1. For core parsers (global lookahead = 1), the `LOOKAHEAD(1)` is already implied;
2. For babel parser (global lookahead = 2), `[LOOKAHEAD(1) <FOO>]` is not faster than `[LOOKAHEAD(2) <FOO>]`. They are basically equivalent.